packages/di/src/lib/injector.test.ts

```diff
@@ -145,4 +146,30 @@ describe('Injector', () => {
 
     injector.get(Service);
   });
+
+  it('should create an instance from a StaticToken factory', () => {
```

Comment:
> Async factories would require a separate asyncInject function. Which could be something worth investigating.

Does it? Aren't you already adding a way here, we can use?

```suggestion
it('should create an instance from an async StaticToken factory', async () => {
    const token = new StaticToken('test-promise', () => new Promise(() => resolve('Hello World'));
    const injector = new Injector();

    const res = injector.get(token);

    expect(await res).to.equal('Hello World');
  });

  it('should create an instance from a StaticToken factory', () => {
```

Reply:
honestly forgot I did that...

Reply:
this also causes some weirdness because .get() would return T | Promise<T>.
I think we would need to add .getAsync() and factoryAsync() to keep things clean.

Reply:
@Phoscur well hell it work lol